wolframalpha/__init__.py

```diff
@@ -1,84 +1,184 @@
+            self.warnings = None
+    
     def _handle_error(self):
-        error = self.tree.find('error')
-        if not error:
+        error_state = self.tree['@error']
+        if error_state == 'false':
             return
-
-        code = error.find('code').text
-        msg = error.find('msg').text
-        tmpl = 'Error {code}: {msg}'
-        raise Exception(tmpl.format(code=code, msg=msg))
-
+        
+        error = self.tree['error']
+        code = error['code']
+        msg = error['msg']
+        template = 'Error {code}: {msg}'
+        raise Exception(template.format(code=code, msg=msg))
+    
     def __iter__(self):
-        return (Pod(node) for node in self.tree.findall('pod'))
-
+        return iter(self.info)
+    
     def __len__(self):
-        return len(self.tree)
-
-    @property
-    def pods(self):
-        return list(iter(self))
-
+        return len(self.info)
+    
     @property
     def results(self):
-        return (pod for pod in self if pod.title=='Result')
+        ''' Get the pods that hold the response to a simple, discrete query. '''
+        return [pod for pod in self.pods if pod.primary or pod.title=='Result']
+    
+    @property
+    def details(self):
+        ''' Get a simplified set of answers with some context. '''
+        return {pod.title: pod.text for pod in self.pods}
 
 class Pod(object):
+    ''' Groups answers and information contextualizing those answers. '''
     def __init__(self, node):
         self.node = node
-        self.__dict__.update(node.attrib)
+        self.error = node['@error']
+        self._handle_error()
+        self.title = node['@title']
+        self.scanner = node['@scanner']
+        self.id = node['@id']
+        self.position = float(node['@position'])
+        self.number_of_subpods = int(node['@numsubpods'])
+        self.subpods = node['subpod']
+        # Allow subpods to be accessed in a consistent way,
+        # as a list, regardless of how many there are.
+        if type(self.subpods) != list:
+            self.subpods = [self.subpods]
+        self.subpods = list(map(Subpod, self.subpods))
+        self.primary = '@primary' in node and node['@primary'] != 'false'
 
+    def _handle_error(self):
+        if self.error == 'false':
+            return
+        
+        error = self.node['error']
+        code = error['code']
+        msg = error['msg']
+        template = 'Error {code}: {msg}'
+        raise Exception(template.format(code=code, msg=msg))
+    
     def __iter__(self):
-        return (Content(node) for node in self.node.findall('subpod'))
-
-    @property
-    def main(self):
-        "The main content of this pod"
-        return next(iter(self))
+        return iter(self.subpods)
+    
+    def __len__(self):
+        return self.number_of_subpods
 
     @property
     def text(self):
-        return self.main.text
+        ''' Simply get the text from each subpod in this pod and return it in a list. '''
```

Comment:
I think this property should be `texts` or `all_texts` and `text` should return what it did before.